### ISSUE TITLE
model.save_raw("json") fails to capture the metrics metadata

### ISSUE THREAD

**Issue by harmbuisman** [dmlc/xgboost#11262]
The config["learner"]["metrics"] values are not exported by the save_raw interface, leading to export raw and loading from it returning a slightly different result, whereas I would have expected to be exactly the same.

Reproduction: this results in an assert error:

```
import xgboost as xgb
import json
import numpy as np

X = np.array([[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]])
y = np.array([0, 1, 0, 1, 0])

train_dmatrix = xgb.DMatrix(X, label=y)

model = xgb.train(
    {},
    train_dmatrix,
    num_boost_round=1,
    evals=[(train_dmatrix, "train")],
)

model_json = model.save_raw("json")
new_model = xgb.Booster(model_file=model_json)

# Compare configurations
cfg = json.loads(model.save_config())
cfg_new = json.loads(new_model.save_config())

print(cfg["learner"]["metrics"], cfg_new["learner"]["metrics"])

# uncomment to verify that this is the only difference
# cfg_new["learner"]["metrics"] = cfg["learner"]["metrics"]

# Expected to be equal
assert cfg_new == cfg

```

**Comment by trivialfis** [comment]
Hi the [save_raw](https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.Booster.save_raw) saves only the model, which we guarantee its stability. If you want the entire thing including hyper parameters, use pickle, which is unstable across versions. See https://xgboost.readthedocs.io/en/stable/tutorials/saving_model.html for details.

**Comment by harmbuisman** [comment]
Hi @trivialfis, thank you for your quick response. 

The description of "save_config" says "Output internal parameter configuration of Booster as a JSON string.". I would assume this something different than "hyperparameter", which I would call external parameters.

"save_model" mentions "Auxiliary attributes of the Python Booster object (such as feature_names) are only saved when using JSON or UBJSON (default) format." which I would assume has all internal booster parameters.

Based on this:
- the documentation could be extended to explain the terminologies "internal parameters", "config", "auxiliary attributes", "hyperparameters" and which of them are included in save_config and save_model
- the field "metrics" in config is unclear and where it comes from. is it a hyperparameter that somehow ended up in the config and shouldn't it be there?
- e.g. : the page mentions "XGBoost accepts user provided objective and metric functions as an extension. These functions are not saved in model file as they are language dependent features." which suggests metrics is user provded, but in the example above I never set a metric. How did it end up in the config?
- the IO page does not mention use cases other than information of internal for the config? in which cases would one use load_config?

**Comment by trivialfis** [comment]
As for 

> The description of "save_config" says "Output internal parameter configuration of Booster as a JSON string.". I would assume this something different than "hyperparameter", which I would call external parameters.

For XGB, anything that used to produce a model, and not part of the model itself, is hyper-parameter.
